src/control/cmd/dmg/pool.go

```diff
@@ -191,6 +192,73 @@ func (r *PoolExcludeCmd) Execute(args []string) error {
 	return err
 }
 
+// PoolExtendCmd is the struct representing the command to Extend a DAOS pool.
+type PoolExtendCmd struct {
+	logCmd
+	ctlInvokerCmd
+	UUID  string `long:"pool" required:"1" description:"UUID of the DAOS pool to extend"`
+	Ranks string `long:"ranks" required:"1" description:"Comma-separated list of ranks to add to the pool"`
+	// Everything after this needs to be removed when pool info can be fetched
```

Comment:
I don't understand this comment. Why are these parameters needed now? There is already a way to query pool info.

Reply:
Sorry I meant to leave a comment explaining why this patch is kind of strange, but apparently I forgot.

The pool extend process will happen in 2 parts. The first part is creating the pool structures on the ranks being added to the existing DAOS pool. In the final version of Online Server Addition the information needed to recreate the pool structures on the new ranks (SCM size, usergroup etc...) will be obtained server side directly from IV.

However that functionality doesn't currently exist. So we discussed the options in the OSA meeting and agreed until we have support to do that down the line that we would add parameters to server addition so that we could pass in the relevant information to create the new pools on the new ranks.